lib/transforms/onnxextension_legalizer.cc

```diff
@@ -392,6 +392,28 @@ static std::vector<Def> ConvertMaximum(const ONNXExtensionInst* ext,
   return {*op0};
 }
 
+static std::vector<Def> ConvertMean(const ONNXExtensionInst* ext,
+                                    IRBuilder* builder) {
+  // Conver to a chain of Mean.
+  auto input = ext->GetOperand(0);
+  auto n = ext->GetNumOfOperands();
+  ConstantBuilder c_builder(ext->GetParent()->GetParent());
+  HLCHECK(n >= 1);
+  if (n == 1) {
+    return {ext->GetOperand(0)};
+  }
+  auto op0 = builder->CreateAdd(ext->GetName(), input, ext->GetOperand(1));
+  for (unsigned i = 2; i < n; ++i) {
+    op0 = builder->CreateAdd(ext->GetName() + std::to_string(i - 1), *op0,
+                             ext->GetOperand(i));
+  }
+  Def mean_div =
+      *c_builder.CreateConstant("NumOfOperands", halo::Type{DataType::FLOAT32},
```

Comment:
mean_div should has the same type as input ("All inputs and outputs must have the same data type.")

input type could be float16, bf16, double, float

for "div" , I think it requires all input be the same type.

Reply:
Ok,I'll fix it